subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
Ok, understood that users must not set the global timeout with this approach because it would set the read response timeout which breaks SSE. However, wouldn't users potentially want to set the read response timeout for regular (non-SSE) API requests? Or do the other timeouts they can still set somehow supersede the problematic read response timeout?